extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java

```diff
@@ -186,6 +191,17 @@ public KubernetesPortBuildItem kubernetesForManagement(
                 managementInterfaceBuildTimeConfig.enabled);
     }
 
+    @BuildStep
+    public void sessionStoreRequest(HttpBuildTimeConfig config, BuildProducer<SessionStoreRequestBuildItem> request) {
+        if (config.sessions.mode == SessionsBuildTimeConfig.SessionsMode.REDIS) {
```

Comment:
I wonder if we should check the [available capabilities](io.quarkus.redis) here; i.e. consume the `io.quarkus.deployment.Capabilities` and then `capabilites.isPresent(Capability.REDIS_CLIENT)`...

Reply:
I originally had that, but the request/response protocol here (which I don't really like, but I don't have a better way at the moment) subsumes that. If the Redis extension isn't present, it won't reply, and we'll get an error.